### ISSUE TITLE
Qunit is not defined errors

### ISSUE THREAD

**Issue by sdhull** [emberjs/ember-mocha#336]
This is a weird one and I'm not sure if there's hope for me 😫 

After converting an in-app-addon to use ember-mocha & chai instead of qunit, _and after successfully running the suite for awhile_, I recently started getting these errors
```
not ok 21 Chrome 72.0 - [1 ms] - TestLoader Failures dummy/tests/lint/templates.template.lint-test: could not be loaded
    ---
        message: >
            QUnit is not defined
        stack: >
            ReferenceError: QUnit is not defined
                at Module.callback (http://localhost:7357/assets/tests.js:5254:3)
                at Module.exports (http://localhost:7357/assets/vendor.js:111:32)
                at requireModule (http://localhost:7357/assets/vendor.js:32:18)
                at TestLoader.require (http://localhost:7357/assets/tests.js:2719:9)
                at TestLoader.loadModules (http://localhost:7357/assets/tests.js:2712:14)
                at Function.load (http://localhost:7357/assets/tests.js:2671:24)
                at http://localhost:7357/assets/test-support.js:27137:18
        Log: |
    ...
not ok 22 Chrome 72.0 - [0 ms] - TestLoader Failures dummy/tests/lint/tests.lint-test: could not be loaded
    ---
        message: >
            QUnit is not defined
        stack: >
            ReferenceError: QUnit is not defined
                at Module.callback (http://localhost:7357/assets/tests.js:5263:3)
                at Module.exports (http://localhost:7357/assets/vendor.js:111:32)
                at requireModule (http://localhost:7357/assets/vendor.js:32:18)
                at TestLoader.require (http://localhost:7357/assets/tests.js:2719:9)
                at TestLoader.loadModules (http://localhost:7357/assets/tests.js:2712:14)
                at Function.load (http://localhost:7357/assets/tests.js:2671:24)
                at http://localhost:7357/assets/test-support.js:27137:18
        Log: |
```

_However_ none of my colleagues nor CircleCI get these errors on this test suite.

I'm thinking there's some sort of universal config I added unknowingly to a ~ dotfile or something? But I've grepped and grepped and come up empty. Any help would be greatly appreciated 🙏

**Comment by Turbo87** [comment]
@sdhull could you try running this with `ember test --server` and check what the code looks like that these stacktraces are referring to?

**Comment by sdhull** [comment]
I think this is what you're asking about:

```
ReferenceError: QUnit is not defined
    at Module.callback (http://localhost:7357/assets/dummy.js:791:3)
    at Module.exports (http://localhost:7357/assets/vendor.js:111:32)
    at requireModule (http://localhost:7357/assets/vendor.js:32:18)
    at TestLoader.require (http://localhost:7357/assets/tests.js:2719:9)
    at TestLoader.loadModules (http://localhost:7357/assets/tests.js:2712:14)
    at Function.load (http://localhost:7357/assets/tests.js:2671:24)
    at http://localhost:7357/assets/test-support.js:27137:18

// dummy.js:788-791
;define("dummy/ember-boundless/tests/addon.lint-test", [], function () {
  "use strict";

  QUnit.module('ESLint | addon');
```

**Comment by Turbo87** [comment]
> After converting an in-app-addon to use ember-mocha & chai instead of qunit

I assume the main app is also using Mocha/Chai?

What `ember-cli-eslint` version are you using?

**Comment by sdhull** [comment]
No I haven't had time to convert the main app to use Mocha/Chai yet 😬
ember-cli-lint in the addon + the main app(s) 5.0.0

**Comment by Turbo87** [comment]
I see, I don't think it will work too have different test frameworks for app and in-repo-addon unfortunately. you will either have to port the app too, or convert the in-repo-addon to a real addon with a dedicated repository.

**Comment by sdhull** [comment]
@Turbo87 the thing that's strange is *I'm* the only one on my team for which the addon suite fails. It runs on CircleCI with no problems, plus runs on my 2 colleagues' machines with no problems.

So my thinking is I have some setting in user-space setting that is injecting these Qunit-style linter checks.

Anyway thanks for asking questions and doing some investigation. Maybe someone else will experience a similar problem and stumble on this issue and offer a different perspective or more clues. 🤞

**Comment by sdhull** [comment]
After having my colleague investigate some more, turns out he has the same block but for some reason it doesn't raise / fail his test run.

Out of curiosity -- how does the in-repo-addon have context of the parent app? I understand that the parent app gets it from `package.json`, from the `"ember-addon"` key...  but in the addon I can't find any reference to the main app.

Like you said 
> I don't think it will work too have different test frameworks for app and in-repo-addon 

I'm curious as to _why_ this is. The addon seems pretty isolated 🤔

**Comment by Turbo87** [comment]
the problem is that ember-cli-eslint detects the test framework from the package.json of the app, and if that is still using QUnit... :wink:

**Comment by sdhull** [comment]
@Turbo87 yeah I guess what I don't understand is how exactly `ember-cli-eslint` detects the test framework of the app. Our addon is not nested under the app and I couldn't find any config or anything pointing to the app(s). (We have a monorepo with 2 apps that use the addon.)

**Comment by Turbo87** [comment]
oh, I thought you were talking about an actual "in-repo-addon" in the `lib` folder of your app repository. if it's a monorepo it should work fine 🤔 

the test framework detection happens at https://github.com/ember-cli/ember-cli-eslint/blob/master/index.js#L20-L27. it should be possible to set a breakpoint there and debug it, if needed.

**Comment by sdhull** [comment]
OK I think I figured this out.

I tried awhile back to migrate from using `in-repo-addon` to yarn workspaces. I ended up abandoning the branch due to difficulties with `ember-fetch` however it left `/node_modules` in our repo root (for me only!). I didn't even notice the dir there because it normally doesn't affect anything but looks like there was `ember-qunit` in there which made version checker (`checker.for('ember-qunit', 'npm').exists()` => `true`) think that it was a qunit project.

So `rm -rf node_modules` in my repo root and everything works properly! Mystery solved! Thanks so much for your help @Turbo87 😀
